### PR TITLE
`reputation`: Refactor and document the package

### PR DESCRIPTION
**Pull request by cthulhu-rider**

### CODE REVIEW

**Review by fyrchik — DISMISSED** [review]

File: reputation/peer.go [hunk 1]
```diff
@@ -2,82 +2,102 @@ package reputation
...
+//
+// Argument MUST NOT be mutated, make a copy first.
+//
+// See also CompareKey.
+func (x *PeerID) SetPublicKey(key []byte) {
+	x.m.SetPublicKey(key)
 }
 
-// ToV2 converts PeerID to NeoFS API v2 reputation.PeerID message.
+// PublicKey return public key set using SetPublicKey.
```
[the remark this answers is not in the document]
Comment (cthulhu-rider) [reply reputation/peer.go:56]: Protocol conversion is completely utilitarian, I want to abstract the user from this level, and create the feeling of working in one code space. This eases the entry threshold and also guarantees compatibility.

File: reputation/trust.go [hunk 2]
```diff
@@ -1,327 +1,416 @@
...
-// 	- version: version.Current();
-//  - manager: nil;
-//  - trust: nil.
-func NewGlobalTrust() *GlobalTrust {
-	gt := GlobalTrustFromV2(new(reputation.GlobalTrust))
-	ver := version.Current()
-	gt.SetVersion(&ver)
-
-	return gt
+// Instances can be created using built-in var declaration.
```
Comment (carpawell) [inline reputation/trust.go:217]: a few words about `Init`?

File: reputation/peer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (carpawell) [reply reputation/peer.go:56]: > create the feeling of working in one code space

not sure that it helps, especially when the doc file contains only examples with `ReadFromV2`

[the remark this answers is not in the document]
Comment (cthulhu-rider) [reply reputation/peer.go:56]: * https://github.com/nspcc-dev/neofs-sdk-go/issues/204

File: reputation/trust.go (hunk 2 above)
Comment (cthulhu-rider) [reply reputation/trust.go]: Added

File: reputation/peer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (carpawell) [reply reputation/peer.go:56]: well, examples are only one point. my general idea: SDK is planned to be used in network communication, NeoFS are about network communication, SDK could not exist without `v2` (`v3`, `v4`, etc), so i do not agree that getter should be documented as "result of setter" when there is no real use case of that flow

[the remark this answers is not in the document]
Comment (cthulhu-rider) [reply reputation/peer.go:56]: Network communication is just an approach of value transmission, RPC technology works like function calling, so it's convenient to be abstracted from the network at all. In total, SDK is not about network only.

**Review by fyrchik — APPROVED** [review]

**Pull request merged**